biggus/__init__.py

```diff
@@ -2287,6 +2346,103 @@ def _sliced_shape(shape, keys):
     return sliced_shape
 
 
+class broadcast(object):
+    def __init__(self, array1, array2):
+        self._a1 = array1
+        self._a2 = array2
+        #: The broadcast shape.
+        self.shape = broadcast.compute_broadcast_shape(array1.shape,
+                                                       array2.shape)
+
+    @staticmethod
+    def compute_broadcast_shape(shape1, shape2):
```

Comment:
Interestingly:

```
>>> import numpy as np
>>> import numpy.lib.stride_tricks
>>> from numpy.lib.stride_tricks import as_strided
>>> strides = (0, 0, 0, 0, 0, 0, 0, 0)
>>> shape = (10000, 100000, 1000000, 10000000, 100000000, 1000000000, 10000000000, 100000000000)
>>> 
>>> np.ndarray(shape, strides=(0,) * len(shape))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: strides is incompatible with shape of requested array and size of buffer
```

I suspect there is an overflow somewhere within numpy causing a problem.

Also, not sure why this would happen if there wasn't a deeper problem (i.e. memory allocation):

```
>>> np.ndarray(shape[:2], strides=(0,) * len(shape[:2]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
MemoryError
```